### PR TITLE
DOC: Fixing EX01 - Added examples

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Towards https://github.com/pandas-dev/pandas/issues/37875

### CODE REVIEW

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -759,6 +772,12 @@ def categories(self) -> Index:
         >>> ser = pd.Series(raw_cat)
         >>> ser.cat.categories
         Index(['b', 'c', 'd'], dtype='object')
+
+        For Categories:
```
Comment (MarcoGorelli) [inline pandas/core/arrays/categorical.py:776]: How about `For Categorical`? likewise below

File: pandas/core/arrays/categorical.py [hunk 2]
```diff
@@ -1492,6 +1530,16 @@ def __array__(self, dtype: NpDtype | None = None) -> np.ndarray:
             A numpy array of either the specified dtype or,
             if dtype==None (default), the same dtype as
             categorical.categories.dtype.
+
+        Examples
+        --------
+
+        >>> cat = pd.Categorical(['a', 'b'], ordered=True)
+
+        The following calls cat.__array__
```
Comment (MarcoGorelli) [inline pandas/core/arrays/categorical.py:1539]: in general for code snippets we probably want double backticks:
```
The following calls ``cat.__array__``
```
(unless it's an argument name, in which case a single backtick is fine. the docs aren't terribly consistent about this admittedly)

**Review by MarcoGorelli — COMMENTED** [review]
nice!

yeah looks like one of the examples might be failing

also, have left some minor comments

**Comment by DeaMariaLeon** [comment]
Thanks for the corrections.

**Comment by DeaMariaLeon** [comment]
@MarcoGorelli - it looks green.

**Review by MarcoGorelli — APPROVED** [review]
nice one - just did a minor fixup but the rest looks good

**Pull request merged**